FlowCrypt/Functionality/Services/GlobalRouter.swift

```diff
@@ -171,8 +171,10 @@ extension GlobalRouter: GlobalRouterType {
             return
         }
 
-        let appContextWithUser = appContext.withSession(session: session, authType: authType, user: user)
-        AppStartup(appContext: appContextWithUser).initializeApp(window: keyWindow)
+        Task {
+            let appContextWithUser = try await appContext.with(session: session, authType: authType, user: user)
+            AppStartup(appContext: appContextWithUser).initializeApp(window: keyWindow)
+        }
```

Comment:
This could conceivably throw. There should be a do/catch and a way to render the problem and retry, eg with an alert.

Reply:
Throwing is handled here, as `func proceed(with appContext: AppContext, session: SessionType)` is marked with `throws`. 
So calling it will need `do/catch` structure, currently it's implemented in `func signIn(appContext: AppContext, route: GlobalRoutingType)` - https://github.com/FlowCrypt/flowcrypt-ios/blob/feature/issue-1348-enterprise-server-refactor/FlowCrypt/Functionality/Services/GlobalRouter.swift#L83

Reply:
Is it possible to actually propagate a thrown exception from inside `Task { }` to the upper method? I was of the understanding that the error will just be ignored if there isn't a `do/catch` inside that Task.

Reply:
Yes, my bad, exception isn't propagated to parent method - it was called from another place when I tried to reproduce this issue. I'll add exception handling here